js/ui/calendar/ui.calendar.js

```diff
@@ -243,28 +248,64 @@ const Calendar = Editor.inherit({
         return dateSerialization.getDateSerializationFormat(value);
     },
 
-    _convertToDate: function(value, optionName) {
+    _createValuesChangeAction: function() {
+        this._valuesChangeAction = this._createActionByOption('onValuesChanged', {
+            excludeValidators: ['disabled', 'readOnly']
+        });
+    },
+
+    _raiseValuesChangeAction: function(value, previousValue) {
+        if(!this._valueChangeAction) {
+            this._createValuesChangeAction();
+        }
+        this._valuesChangeAction(this._valuesChangeArgs(value, previousValue));
+    },
+
+    _valuesChangeArgs: function(value, previousValue) {
+        return {
+            value: value,
+            previousValue: previousValue,
+            event: this._valuesChangeEventInstance
+        };
+    },
+
+    _saveValuesChangeEvent: function(e) {
+        this._valuesChangeEventInstance = e;
+    },
+
+    _convertToDate: function(value) {
         return dateSerialization.deserializeDate(value);
     },
 
     _dateValue: function(value, event) {
+        const optionName = Array.isArray(value) ? 'values' : 'value';
         if(event) {
             if(event.type === 'keydown') {
                 const cellElement = this._view._getContouredCell().get(0);
                 event.target = cellElement;
             }
-            this._saveValueChangeEvent(event);
+            if(optionName === 'value') {
+                this._saveValueChangeEvent(event);
```

Comment:
Do we really need a separate event?

Reply:
now onValueChanged is used for both `value` and `values`